sparse_ho/ho.py

```diff
@@ -13,34 +13,41 @@ def grad_search(
 
     Parameters
     ----------
-    algo: TODO
-    criterion: TODO
-    model: TODO
+    algo: Algo object
+        algorithm used to compute hypergradient.
+    criterion: Criterion object
```

Comment:
instance of Criterion

btw if Algo and Criterion are classes that are not meant to be used directly they are base classes and should be called BaseAlgo and BaseCriterion

Reply:
It's just for me to understand, it is BaseAlgo even if we do not have a class algorithm on which depends all our algorithms?

Reply:
I think it would be clearer if you had BaseAlgo, BaseCriterion, BaseModel. It makes it clear for someone who wants to implement is own criterion our algo, which methods he must implement inside

Reply:
Yes, you're right, I changed it and created a new PR without these "old" commits.

Reply:
Ok, maybe close this one to avoid duplication